core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -231,6 +225,16 @@ private Object handleTxCommand(TxInvocationContext ctx, TransactionBoundaryComma
          }
       } else {
          if (currentTopologyId() > command.getTopologyId()) {
+            //in some cases, the node will not retry the transaction in this node and then it leaks.
+            //if it is safe, we rollback the transaction. If the coordinator retries on this node, no side effect is created.
```

Comment:
I feel like this is the wrong way to fix it, and we should make sure that if the retry doesn't happen on all the nodes that got the original command, then at least the commit/rollback happens on all of them. Or is this specifically related to the nodes being in different partitions during commit?